source/extensions/filters/http/istio_stats/istio_stats.cc

```diff
@@ -58,17 +58,37 @@ absl::string_view extractMapString(const ProtobufWkt::Struct& metadata,
   return extractString(it->second.struct_value(), key);
 }
 
+absl::optional<Istio::Common::WorkloadMetadataObject> extractEndpointMetadata(
+    const StreamInfo::StreamInfo& info) {
+  auto upstream_info = info.upstreamInfo();
+  auto upstream_host = upstream_info ? upstream_info->upstreamHost() : nullptr;
+  if (upstream_host && upstream_host->metadata()) {
+    const auto& filter_metadata = upstream_host->metadata()->filter_metadata();
+    const auto& it = filter_metadata.find("istio");
+    if (it != filter_metadata.end()) {
+      const auto& workload_it = it->second.fields().find("workload");
+      if (workload_it != it->second.fields().end()) {
+        return Istio::Common::convertEndpointMetadata(
+            workload_it->second.string_value());
+      }
+    }
+  }
+  return {};
+}
+
 enum class Reporter {
   ClientSidecar,
   ServerSidecar,
+  ServerGateway,
```

Comment:
What is a ServerGateway? Is ServerGateway for ingress/egress gateway?

Reply:
It's "waypoint" proxy in Ambient mode, which is a server proxy shared between backends.
Basically, it's a sidecar that is shared between deployment pods.

Reply:
This is usually called internal gateway, but "server" sort of implies internal since it's not cluster gateway.

Reply:
can you add some comment for this?